thold_functions.php

```diff
@@ -2233,11 +2233,14 @@ function thold_check_threshold(&$thold_data) {
 
 			thold_debug('Threshold HI / Low check breached HI:' . $thold_data['thold_hi'] . '  LOW:' . $thold_data['thold_low'] . ' VALUE:' . $thold_data['lastread']);
 
-			$thold_data['thold_fail_count']++;
+			if (!$maint_dev) {
+				$thold_data['thold_fail_count']++;
+			}
 			$thold_data['thold_alert'] = ($breach_up ? STAT_HI : STAT_LO);
 
-			/* Re-Alert? */
-			$ra = ($thold_data['thold_fail_count'] > $trigger && $thold_data['repeat_alert'] != 0 && $thold_data['thold_fail_count'] % $thold_data['repeat_alert'] == 0);
+			/* Re-Alert? Exclude initial trigger count */
+			$ra = ($thold_data['thold_fail_count'] > $trigger && $thold_data['repeat_alert'] != 0
+					&& ($thold_data['thold_fail_count'] - $trigger) % $thold_data['repeat_alert'] == 0);
```

Comment:
The second half of this expression uses the modulo but not "minus the $trigger", that is automatic for modulo.  So, it's redundant.

Reply:
I am not sure what needs to be changed.  I will try to explain the scenario and intent.

With one test scenario (probably not reasonable), the initial notify warn count was 3 and the repeat notify was 2.  The warning was sent at the expected interval 3 and the repeat notify the next interval 4, not 2 intervals later.

`current warn count 4 divided by repeat count 2 = 2 with no remainder.  Repeat notify sent`

I thought the repeat notify should be 2 intervals after the first notify and so the initial count is subtracted from the current.

`current warn count 4 minus initial warn notify count 3 = 1`
`previous value 1 divided by repeat count 2 = remainder 1.  No notify at count 4`

`current warn count 5 minus initial warn notify count 3 = 2`
`previous value 2 divided by repeat count 2 = remainder 0.  Repeat notify`

`nwwwwwn`
` 12345`
`   WR< repeat notify not expected`
`   W R with change to calculation`

The following was my attempt to subtract the initial notify count from the current count before determining the repeat interval.

`($thold_data['thold_fail_count'] - $trigger) % $thold_data['repeat_alert'] == 0)`